tests/integration/extenders/NotificationTest.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Tests\integration\extenders;
+
+use Flarum\Extend;
+use Flarum\Notification\Blueprint\BlueprintInterface;
+use Flarum\Notification\Notification;
+
+class NotificationTest extends \Flarum\Tests\integration\TestCase
+{
+    /**
+     * @test
+     */
+    public function notification_type_doesnt_exist_by_default()
+    {
+        $this->assertArrayNotHasKey('customNotificationType', Notification::getSubjectModels());
+    }
+
+    /**
+     * @test
+     */
+    public function notification_type_exists_if_added()
```

Comment:
More tests that would be nice to have could cover that the serializer is properly registered, and that preferences are properly saved.

Reply:
so we just need to do some checks on the `$subjectSerializers` static property of the `NotificationSerializer` class and `$preferences` static property of the `User` model, correct ?

Reply:
I think so. We could even throw in some reflection for Serializer but I'm not sure that's necessary at the moment

Reply:
well, those two properties are protected, we'd have to create static methods to read them, is that fine when the only use is for tests ?

Reply:
Hmm I didn't check for that. In that case, let's leave it as is.